helix-term/src/commands.rs

```diff
@@ -1150,13 +1151,36 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
                 .to_string(),
         );
     }
+
     for sel in paths {
         let p = sel.trim();
-        if !p.is_empty() {
-            if let Err(e) = cx.editor.open(&PathBuf::from(p), action) {
-                cx.editor.set_error(format!("Open file failed: {:?}", e));
+        if p.is_empty() {
+            continue;
+        }
+
+        if let Ok(url) = Url::parse(p) {
+            // file:// urls are opened inside helix
+            if url.scheme() != "file" {
+                let commands = open::commands(url.as_str());
+                cx.jobs.callback(async {
+                    for cmd in commands {
+                        let mut command = tokio::process::Command::new(cmd.get_program());
+                        command.args(cmd.get_args());
+                        if command.output().await.is_ok() {
+                            return Ok(job::Callback::Editor(Box::new(|_| {})));
+                        }
+                    }
+                    Ok(job::Callback::Editor(Box::new(move |editor| {
+                        editor.set_error("Open file failed: no command found")
+                    })))
+                });
```

Comment:
Extracted into separate function but wasn't sure what the best place for it would be (until we find a use-case for re-using it). So for now is still lives in `helix-term/src/commands.rs`.

Reply:
its ok to keep there for now. I would eventually like to move this to helix-core if we endup needing that elsewhere but that is currently not possible since you can't create callbacks there but I think its fine to do that once we actually need it outside of helix-term